src/Shared/FileMatcher.cs

```diff
@@ -1475,18 +1475,21 @@ static string[] CreateArrayWithSingleItemIfNotExcluded(string filespecUnescaped,
             {
                 foreach (string excludeSpec in excludeSpecsUnescaped)
                 {
-                    //  The FileMatch method always creates a Regex to check if the file matches the pattern
-                    //  Creating a Regex is relatively expensive, so we may want to avoid doing so if possible
-                    Result match = FileMatch(excludeSpec, filespecUnescaped);
+
+                    if (FileUtilities.PathsEqual(filespecUnescaped, excludeSpec))
```

Comment:
This block seems worthy of a comment explaining. Agreed with deleting the old comment since it was decent advice that was promptly not followed :)